Sources/Build/BuildPlan.swift

```diff
@@ -388,8 +388,15 @@ public final class ProductBuildDescription {
         args += ["-L", buildParameters.buildPath.asString]
         args += ["-o", binary.asString]
         args += ["-module-name", product.name]
-        args += dylibs.map({ "-l" + $0.product.name })
-
+        
+        func dylibsIn(product: ProductBuildDescription) -> [ProductBuildDescription] {
+            return product.dylibs.reduce([product], {$0 + dylibsIn(product: $1)})
```

Comment:
Minor nitpick: Can we have one space inside the closure to match the project style? `{ $0 + dylibsIn(product: $1) }`? Same on line 398.